### PR TITLE
TGUI list conversions + bug fixes

### LINKED ISSUE #63307: NtOS error when inputting a revolver into a custom vending machine (And potentially other items)
## Round ID: 
[174423](https://scrubby.melonmesa.com/round/174423)

## Reproduction:

When setting up a custom vending machine, I tagged a 357 revolver for 500cr and put it into the machine. However, trying to access the machine resulted in a NtOS bug as follows. Trying to re-access the machine yielded the same error, and another player tried to access the vending but also saw the error. On a local build I tried this again and got the same issue. Also seems to happen when I inserted a tagged wrench.

TypeError: Unable to get property 'amount' of undefined or null reference
   at s (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:1531102)
   at Ie (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179474)
   at Anonymous function (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179742)
   at Le (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179720)
   at De (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:180768)
   at Te (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:180607)
   at Le (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179584)
   at Te (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:180573)
   at Le (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179584)
   at Anonymous function (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179742)
User Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.2; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729)
State: {"ckey":"fastmapler","interface":"Vending","window":{"key":"[0x2006310]-main","size":null,"fancy":1,"locked":0}}

In short:
1. Build a Custom Vendor using a Custom Vendor machine board and resupply canister
2. Link the vending machine to you by using your ID card on it
3. Obtain a 357 revolver, and use a price tagger on it to set a price
4. Insert the revolver into the machine, and attempt to open the vending machine's interface

### LINKED ISSUE #63629: Succumb TGUI input requires text to succumb
If you try and succumb, the TGUI input that appears requires some sort of input to it, whereas it used to accept nothing

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
- Converts more inputs to TGUI. Possibly all user-facing input lists in the game.
- Did any surrounding text/number inputs as well
- Added null choice support so users can press cancel.
- Added some misc TGUI input fixes
- Fixed custom vendors while I was there

I refactored a lot of code while just poking around.
Primarily, usage of .len in files where I was already working on lists.
Some code was just awful - look at guardian.dm and its non use of early returns
If there are any disputes, I can revert it just fine, those changes are not integral to the PR.

## Why It's Good For The Game
Fixes #63629
Fixes #63307
Fixes custom vendors /again/
Text input is more performant.
Part of a long series of TGUI conversion to make the game more visually appealing

## Changelog

:cl:
refactor: The majority of user facing input lists have been converted to TGUI.
refactor: Tgui text inputs now scale with entered input.
fix: Many inputs now properly accept cancelling out of the menu.
fix: Fixes an edge case where users could not press enter on number inputs.
fix: Custom vendor bluescreen.
fix: You can now press ENTER on text inputs without an entry to cancel.
/:cl:

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1266,39 +1266,39 @@
...
-		var/result = input(usr, "Choose the transformation to apply","Transform Mod") as null|anything in list("Scale","Translate","Rotate")
+		var/result = tgui_alert(usr, "Choose the transformation to apply", "Transform Mod", list("Scale","Translate","Rotate"))
 		var/matrix/M = transform
 		if(!result)
 			return
 		switch(result)
 			if("Scale")
-				var/x = input(usr, "Choose x mod","Transform Mod") as null|num
-				var/y = input(usr, "Choose y mod","Transform Mod") as null|num
+				var/x = tgui_input_number(usr, "Choose x mod"," Transform Mod", min_value = null)
```
Comment (JohnFulpWillard) [inline code/game/atoms.dm:1281]:
```suggestion
				var/x = tgui_input_number(usr, "Choose x mod", "Transform Mod", min_value = null)
```

File: code/controllers/admin.dm [hunk 2]
```diff
@@ -70,22 +70,22 @@ INITIALIZE_IMMEDIATE(/obj/effect/statclick)
...
+
 	for (var/datum/controller/controller in world)
 		if (istype(controller, /datum/controller/subsystem))
 			continue
 		controllers["[controller] (controller.type)"] = controller //we use an associated list to ensure clients can't hold references to controllers
 		controller_choices += "[controller] (controller.type)"
-	
-	var/datum/controller/controller_string = input("Select controller to debug", "Debug Controller") as null|anything in controller_choices
+
+	var/datum/controller/controller_string = tgui_input_list(src, "Select controller to debug", "Debug Controller", controller_choices)
```
Comment (LemonInTheDark) [inline code/controllers/admin.dm:83]: This is debug tooling, please do not convert it

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -1711,7 +1709,7 @@
 		if(answer == "yes")
 			log_query_debug("[usr.key] | Reported a server hang")
 			if(tgui_alert(usr, "Had you just press any admin buttons?", "Query server hang report", list("Yes", "No")) == "Yes")
-				var/response = input(usr,"What were you just doing?","Query server hang report") as null|text
+				var/response = tgui_input_text(usr, "What were you just doing?", "Query server hang report")
```
Comment (LemonInTheDark) [inline code/modules/admin/topic.dm:1712]: also a bit of a bad idea (Though I'm not sure this has ever been used)

File: code/__HELPERS/files.dm [hunk 4]
```diff
@@ -20,7 +20,7 @@ GLOBAL_VAR_INIT(fileaccess_timer, 0)
 		if(path != root)
 			choices.Insert(1,"/")
 
-		var/choice = input(src,"Choose a file to access:","Download",null) as null|anything in sort_list(choices)
+		var/choice = tgui_input_list(src, "File to access", "Download", sort_list(choices))
```
Comment (LemonInTheDark) [inline code/__HELPERS/files.dm:23]: This also should not be converted

File: code/__HELPERS/atoms.dm [hunk 5]
```diff
@@ -248,10 +248,9 @@ rough example of the "cone" made by the 3 dirs checked
 ///Returns a chosen path that is the closest to a list of matches
 /proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
 	if (value == FALSE) //nothing should be calling us with a number, so this is safe
-		value = input("Enter type to find (blank for all, cancel to cancel)", "Search for type") as null|text
+		value = tgui_input_text(usr, "Enter type to find (blank for all)", "Search For Type")
```
Comment (LemonInTheDark) [inline code/__HELPERS/atoms.dm:251]: This proc is used as a part of vv, please keep it clear

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Touches debug procs 🗞️

File: code/game/objects/items/cards_ids.dm [hunk 6]
```diff
@@ -524,7 +522,7 @@
 		registered_account.bank_card_talk(span_warning("内部服务器错误"), TRUE)
```
Comment (jlsnow301) [inline code/game/objects/items/cards_ids.dm:522]: wow

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 7]
```diff
@@ -453,38 +453,41 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
...
-					if("miner")
-						to_chat(src, span_holoparasite("<font color=\"[G.guardiancolor]\"><b>[G.real_name]</b></font> has appeared!"))
-				guardians -= G
-				if(!guardians.len)
-					remove_verb(src, /mob/living/proc/guardian_reset)
-			else
-				to_chat(src, span_holoparasite("There were no ghosts willing to take control of <font color=\"[G.guardiancolor]\"><b>[G.real_name]</b></font>. Looks like you're stuck with it for now."))
-		else
-			to_chat(src, span_holoparasite("You decide not to reset [guardians.len > 1 ? "any of your guardians":"your guardian"]."))
-	else
```
Comment (jlsnow301) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:487]: pain

Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm:487]: I am sorry for your loss

File: code/modules/antagonists/brother/brother.dm [hunk 8]
```diff
@@ -99,7 +99,7 @@
 		candidates[L.mind.name] = L.mind
 
 	var/choice = input(admin,"Choose the blood brother.", "Brother") as null|anything in sort_names(candidates)
-	if(!choice)
+	if(isnull(choice))
```
Comment (LemonInTheDark) [inline code/modules/antagonists/brother/brother.dm:102]: Why this change? are you trying to protect against "falsey" inputs getting ignored?

Comment (jlsnow301) [reply code/modules/antagonists/brother/brother.dm:102]: just trying to keep with using the macros - i don't think there will be many cases where we'll get a falsey value, but it won't hurt. explosions.dm or that view range selection, maybe.

Comment (jlsnow301) [reply code/modules/antagonists/brother/brother.dm:102]: shorter answer, i'm "being extra"

Comment (LemonInTheDark) [reply code/modules/antagonists/brother/brother.dm:102]: Keep in mind, just checking for falsey rather then null is fine in most places. As an input sanity it's a good pattern (Someone somewhere will care about an empty string), but outside of that it's perfectly fine to be general

**Comment by jlsnow301** [comment]
This is ready. Sorry that's it's massive

**Comment by derekhearst** [comment]
Seems to have broken the ability to change the medical cyborgs injector, you can't change what you want to inject and no gui comes up.

**Comment by jlsnow301** [comment]
> Seems to have broken the ability to change the medical cyborgs injector, you can't change what you want to inject and no gui comes up.

thx, will look into this
